runtime/lua/vim/lsp/util.lua

```diff
@@ -1777,12 +1786,13 @@ function M.locations_to_items(locations, offset_encoding)
       local pos = temp.start
       local row = pos.line
       local line = lines[row] or ''
+      local text = (opts.format or function() return line end)(line, temp.location)
```

Comment:
should we just give control of the entire quickfix item? Then the callback could be named `on_item`

Reply:
> so something like function on_item(filename, lnum, col, text)

can `on_item` take the already-constructed qf item? That will save hassle for people setting the callback. Then the callback can just set whatever field it's interested in and pass it along.

> 0- or 1-based indexing for that? 

whatever we do elsewhere in lsp.lua.